### PR TITLE
Apportion when a fixed value coupon

### PR DESCRIPTION
**Pull request by ryanmitchell**

### CODE REVIEW

File: cartconditions/Coupon.php [hunk 1]
```diff
@@ -92,12 +93,28 @@ public function beforeApply()
...
+            $applicableItems = self::$applicableItems;
+            $applicableItemsTotal = Cart::content()->sum(function ($cartItem) use ($applicableItems) {
+                if (!$applicableItems->contains($cartItem->id))
+                    return 0;
+
+                return $cartItem->subtotalWithoutConditions();
+            });
+
+            $value = ($this->target->subtotalWithoutConditions() / $applicableItemsTotal) * (float)$value;
+        }
```
Comment (sampoyigi) [inline cartconditions/Coupon.php]: Works nicely! 

Move this logic into a separate method. I don't think we need line 114, the `calculateValue` method is not defined anywhere.

**Review by sampoyigi — CHANGES_REQUESTED** [review]

File: cartconditions/Coupon.php (hunk 1 above)
Comment (sampoyigi) [reply cartconditions/Coupon.php]: Just to add, this `if (!$applicableItems->contains($cartItem->id))` throws an error when `$applicableItems` is null.

Comment (ryanmitchell) [reply cartconditions/Coupon.php]: Should be changed

**Comment by sampoyigi** [comment]
Hmmm, it's starting to look like it should be supported by the underlying CartCondition class. Let me see if I can make the Tax condition approach work with the same calculation.

**Comment by ryanmitchell** [comment]
😀 I think this approach works well. There are other offer types that are going to need custom logic and using value like this makes it easy enough to do.

**Comment by sampoyigi** [comment]
Of course, there's nothing wrong with the approach I'm too obsessed with refactoring 😅  I'll merge now, thank you!

**Pull request merged**